DEPRECATION.md

```diff
@@ -0,0 +1,101 @@
+ - deprecation period;
+ - how "internal" they are perceived.
+
+If a batch proposal fails, it's fine to split it into multiple proposals. In
+general, one proposal per module would seem too fine a granularity.
+
+## Process details
+
+The key idea of this process is to separate the decision-making stage (for which
+CLC oversight is required) from the implementation stage (which happens over
+time, without need for further input from the CLC).
+
+### Decision stage
+
+* The proposer [creates an issue](https://github.com/haskell/core-libraries-committee/issues/new)
+  identifying the definitions to be deprecated and removed, referencing this
+  policy.  The proposal should include an estimate of the number of Hackage
+  packages that will be impacted by the change, and the duration of the proposed
+  deprecation period.  However, an implementation MR or full impact assessment
+  is not required at this stage.
+
+* CLC members or other interested parties comment on the issue. In particular,
+  they may raise concerns for specific functions, saying "we actually want those
+  in `base`", and propose a new home for those functions (which module exactly).
+
+* At least two weeks after the issue creation, the proposer updates the issue
+  description with the final version of the proposal following the discussion,
+  and asks the CLC to vote.
+
+* The CLC votes on the proposal. If it is rejected, the process stops. If it is
+  accepted, the GHC developers may proceed to the implementation section below.
+
+### Implementation stage
+
+All changes during the implementation stage are subject to the usual GHC MR
+review process. The MRs should link to the CLC issue agreeing to the
+deprecation. The CLC are welcome, but not obliged, to comment on implementation MRs.
+
+* The implementer adds Haddock comments and `DEPRECATED` pragmas to the affected
+  definitions. documenting that they are scheduled for removal. The Haddock
+  comments should include:
+
+   * the planned last GHC and/or `base` version that will allow to import this module; and
+
+   * the migration strategy (different module, different package, different functions...),
+     or if there is no direct migration possible, an explanation why.
+
+* The deprecation period starts when a `base` release including deprecation
+  comments is published.
+
+* During the deprecation period, if it comes to light that there is a compelling
+  reason to reverse the deprecation, e.g. due to feedback from users, this
+  should be brought to the attention of the GHC developers via the GHC issue
+  tracker. They may choose to extend the deprecation period or solicit CLC
+  feedback.
+
+* Deprecated definitions can be removed by the GHC developers in any major
+  release of `base` published after the deprecation period has elapsed.  The
+  `base` changelog should mention the definitions that have been removed, and
+  the migration strategy.
+
```

Comment:
```suggestion

* GHC developers maintain a list of modules that are
    - candidates for deprecation (e.g. GHC.* namespace)
    - either fully or partially accepted to be deprecated by a CLC vote (document: deprecation cycle)
    - currently in active deprecation phase shipped by a given base version (document: first possible major GHC version that may drop the definitions)
    - fully deprecated and removed from base (document: first base version that lacks the definitions)

```

Reply:
@bgamari given your https://github.com/haskell/core-libraries-committee/issues/218#issuecomment-1921732367, does this sound okay to you?